tests/ui/no_std/simple-runs.rs

```diff
@@ -0,0 +1,33 @@
+//! Check that `no_std` binaries can link and run without depending on `libstd`.
+
+//@ run-pass
+//@ compile-flags: -Cpanic=abort
+//@ ignore-wasm different `main` convention
+
+#![no_std]
+#![no_main]
+
+use core::ffi::{c_char, c_int};
+use core::panic::PanicInfo;
+
+// Need to link libSystem on Apple platforms, otherwise the linker fails with:
+// > ld: dynamic executables or dylibs must link with libSystem.dylib
+//
+// With the new linker introduced in Xcode 15, the error is instead:
+// > Undefined symbols: "dyld_stub_binder", referenced from: <initial-undefines>
+//
+// This _can_ be worked around by raising the deployment target with
+// MACOSX_DEPLOYMENT_TARGET=13.0, though it's a bit hard to test that while
+// still allowing the test suite to support running with older Xcode versions.
+#[cfg_attr(target_vendor = "apple", link(name = "System"))]
+extern "C" {}
```

Comment:
We _could_ just always link `libSystem` internally in `rustc`, but I'm not sure that's a good idea, since as noted, it _is_ possible to create a fully `no_std` binary that doesn't link `libSystem`, it just requires the user to do some work.

Maybe better would be to document the limitation somewhere (where?)? Note that `libc` does the linking internally too (linking to `libc.dylib` also links to `libSystem.dylib`), so any code that does `#![no_std]` but uses `libc` works without this.